indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -872,4 +873,58 @@ private void taskComplete(
     // Notify interested parties
     taskRunnerWorkItem.setResult(taskStatus);
   }
+
+  public Collection<ZkWorker> findLazyworkers(Predicate<ZkWorker> isLazyWorker, int excessWorkers)
+  {
+    // status lock is used to prevent any tasks being assigned to the worker while we mark it lazy
+    synchronized (statusLock) {
+      Iterator<String> iterator = zkWorkers.keySet().iterator();
+      while (iterator.hasNext()) {
+        String worker = iterator.next();
+        ZkWorker zkWorker = zkWorkers.get(worker);
+        try {
+          if (getAssignedTasks(zkWorker.getWorker()).isEmpty() && isLazyWorker.apply(zkWorker)) {
+            log.info("Adding Worker[%s] to lazySet!", zkWorker.getWorker().getHost());
+            lazyWorkers.put(worker, zkWorker);
+            iterator.remove();
```

Comment:
I feel that modifying zkWorkers directly is a bit dangerous, since it's being used both to know the set of announced workers and the set of active workers. Is it possible to only be the set of announced workers and rely on taking the difference between zkWorkers and lazy workers whenever we assign a taks?

I am worried that a zkWorker that disappears and then re-appears in ZK may inadvertantly end up in both sets.